File: bitz-php-sdk/BitzApi.php
<?php
/**
 * Created by Bitz
 * Date 2019-10-12
 */

class BitzApi
{

	protected $apiKey = '';
	protected $secretKey = '';
	protected $tradePwd = '';
	protected $url = '';


	public function __construct($options = null)
	{
		try {
			if (is_array($options))
			{
				foreach ($options as $option => $value)
				{
					$this->$option = $value;
				}
			}
			else
			{
				return false;
			}
		}
		catch (PDOException $e) {
			throw new Exception($e->getMessage());
		}
	}


	/**
	 * 获取合约交易市场列表
	 * @param contractId 选填 int
	 * @return array
	 */
	public function getContractCoin(int $contractId = 0){
		$url = $this->url.'/Market/getContractCoin?contractId='.intval($contractId);
		return $this->httpRequest($url);
	}


	/**
	 * 获取合约交易最新行情
	 * @param contractId 选填 int
	 * @return array
	 */
	public function getContractTickers(int $contractId = 0){
		$url = $this->url.'/Market/getContractTickers?contractId='.intval($contractId);
		return $this->httpRequest($url);
	}


	/**
	 * 获取合约交易的市场深度
	 * @param contractId 必填 int
	 * @param depth 选填 string
	 * @return array
	 */
	public function getContractOrderBook(array $parms){
		$url = $this->url.'/Market/getContractOrderBook?'.http_build_query($parms);
		return $this->httpRequest($url);
	}


	/**
	 * 获取某个合约的成交历史
	 * @param contractId 必填 int
	 * @param pageSize 选填 int
	 * @return array
	 */
	public function getContractTradesHistory(array $parms){
		$url = $this->url.'/Market/getContractTradesHistory?'.http_build_query($parms);
		return $this->httpRequest($url);
	}


	/**
	 * 获取某个合约市场的K线数据
	 * @param contractId 必填 int
	 * @param type 必填 string
	 * @param size 选填 int
	 * @return array
	 */
	public function getContractKline(array $parms){
		$url = $this->url.'/Market/getContractKline?'.http_build_query($parms);
		return $this->httpRequest($url);
	}


	/**
	 * 获取合约账户权益(资产)
	 * @return array
	 */
	public function getContractAccountInfo(){
		$sendData  = $this->getData();
		$url = $this->url.'/Contract/getContractAccountInfo';
		return $this->httpRequest($url,$sendData);
	}


	/**
	 * 下单 做多/做空
	 * @param contractId 必填  int
	 * @param price      选填  float
	 * @param amount     必填  int
	 * @param leverage   必填  float
	 * @param direction  必填  int
	 * @param type       必填  string
	 * @param isCross    必填  int
	 * @return array
	 */
	public function addContractTrade(array $parms){
		$sendData  = $this->getData($parms);
		$url = $this->url.'/Contract/addContractTrade';
		return $this->httpRequest($url,$sendData);
	}


	/**
	 * 撤销委托单
	 * @param entrustSheetId 必填 int
	 * @return array
	 */
	public function cancelContractTrade(array $parms){
		$sendData  = $this->getData($parms);
		$url = $this->url.'/Contract/cancelContractTrade';
		return $this->httpRequest($url,$sendData);
	}


	/**
	 * 获取当前委托单
	 * @param contractId 必填 int
	 * @return array
	 */
	public function getContractOrder(array $parms){
		$sendData  = $this->getData($parms);
		$url = $this->url.'/Contract/getContractOrder';
		return $this->httpRequest($url,$sendData);
	}


	/**
	 * 获取当前未平仓位
	 * @param contractId 必填 int
	 * @return array
	 */
	public function getContractActivePositions(array $parms){
		$sendData  = $this->getData($parms);
		$url = $this->url.'/Contract/getContractActivePositions';
		return $this->httpRequest($url,$sendData);
	}


	/**
	 * 获取历史委托单详情
	 * @param contractId   必填  int
	 * @param pageSize     选填  int
	 * @param page         选填  int
	 * @return array
	 */
	public function getContractMyHistoryTrade(array $parms){
		$sendData  = $this->getData($parms);
		$url = $this->url.'/Contract/getContractMyHistoryTrade';
		return $this->httpRequest($url,$sendData);
	}


	/**
	 * 获取我的已平仓位
	 * @param contractId   必填  int
	 * @param pageSize     选填  int
	 * @param page         选填  int
	 * @return array
	 */
	public function getContractMyPositions(array $parms){
		$sendData  = $this->getData($parms);
		$url = $this->url.'/Contract/getContractMyPositions';
		return $this->httpRequest($url,$sendData);
	}


	/**
	 * 获取单个或多个委托单明细
	 * @param entrustSheetIds  必填  string
	 * @return array
	 */
	public function getContractOrderResult(array $parms){
		$sendData  = $this->getData($parms);
		$url = $this->url.'/Contract/getContractOrderResult';
		return $this->httpRequest($url,$sendData);
	}


	/**
	 * 获取某个委托的成交明细
	 * @param entrustSheetId   必填  int
	 * @return array
	 */
	public function getContractTradeResult(array $parms){
		$sendData  = $this->getData($parms);
		$url = $this->url.'/Contract/getContractTradeResult';
		return $this->httpRequest($url,$sendData);
	}


	/**
	 * 获取我的成交历史
	 * @param contractId   必填  int
	 * @param pageSize     选填  int
	 * @param page         选填  int
	 * @return array
	 */
	public function getContractMyTrades(array $parms){
		$sendData  = $this->getData($parms);
		$url = $this->url.'/Contract/getContractMyTrades';
		return $this->httpRequest($url,$sendData);
	}


	//获取带签名的参数
	protected function getData($data = null){
		$baseArr = array(
				'apiKey'      =>  $this->apiKey,
				'timeStamp'   =>  time(),
				'nonce'		  =>  $this->getRandomString(6),
				);

		if(isset($data)){
			$sendData = array_merge($baseArr,$data);
		}else{
			$sendData = $baseArr;
		}
		$sendData['sign'] = $this->getSign($sendData);
		return $sendData;
	}


	//参数签名
	protected function getSign($data){
		ksort($data);
		foreach( $data as $k=>$v)
		{
			if(!$v){
				unset($data[$k]);
			}
		}

		$dataStr = '';
		foreach ($data as $k => $v)
		{

			$dataStr .= $k.'='.$v."&";
		}
		return md5(trim($dataStr,"&").$this->secretKey);
	}


	//获取随机字符串nonce
	protected function getRandomString($len, $chars=null){
		if (is_null($chars)){
			$chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
		}

		for ($i = 0, $str = '', $lc = strlen($chars)-1; $i < $len; $i++){
			$str .= $chars[mt_rand(0, $lc)];
		}
		return $str;
	}
	

	//HTTP请求
	protected function httpRequest($url,$data = null){
		$curl = curl_init();
		curl_setopt($curl, CURLOPT_URL, $url);
		curl_setopt($curl, CURLOPT_SSL_VERIFYPEER, FALSE);
		curl_setopt($curl, CURLOPT_SSL_VERIFYHOST, FALSE);
		curl_setopt($curl, CURLOPT_TIMEOUT, 100);
		if(!empty($data)){
			curl_setopt($curl,CURLOPT_POST,1);
			curl_setopt($curl,CURLOPT_POSTFIELDS,$data);
		}
		curl_setopt($curl, CURLOPT_RETURNTRANSFER, 1);
		$output = curl_exec($curl);
		curl_close($curl);
		return json_decode($output,1);
	}
}

// $bitz = new BitzApi();
//获取合约所有市场
// $res = $bitz->getContractCoin();
//获取合约单个市场数据
// $contractId = 1;
// $res = $bitz->getContractCoin($contractId);

//获取合约市场最新行情
// $contractId = 101;
// $res = $bitz->getContractTickers($contractId);

//获取合约市场深度
// $parms = array('contractId'=>101,'depth'=>'5');
// $res = $bitz->getContractOrderBook($parms);

//获取合约市场成交历史
// $parms = array('contractId'=>101,'pageSize'=>'10');
// $res = $bitz->getContractTradesHistory($parms);

//获取合约市场k线数据
// $parms = array('contractId'=>101,'type'=>'5m','size'=>100);
// $res = $bitz->getContractKline($parms);

//获取我的账户权益
// $res = $bitz->getContractAccountInfo();

//下单委托
// $parms = array('contractId'=>101,'price'=>8200.01,'amount'=>100,'leverage'=>10,'direction'=>1,'type'=>'limit','isCross'=>-1);
// $res = $bitz->addContractTrade($parms);

//取消委托
// $parms = array('entrustSheetId'=>15484932);
// $res = $bitz->cancelContractTrade($parms);

//获取我的当前委托
// $parms = array('contractId'=>101);
// $res = $bitz->getContractOrder($parms);

//获取我的未平仓位
// $parms = array('contractId'=>101);
// $res = $bitz->getContractActivePositions($parms);

//获取我的委托历史
// $parms = array('contractId'=>101,'page'=>1,'pageSize'=>50);
// $res = $bitz->getContractMyHistoryTrade($parms);

//获取我的已平仓位
// $parms = array('contractId'=>101,'page'=>1,'pageSize'=>50);
// $res = $bitz->getContractMyPositions($parms);

//获取单个或多个委托单的委托明细
// $parms = array('entrustSheetIds'=>'15485465,15484932,14023097');
// $res = $bitz->getContractOrderResult($parms);

//获取某个委托单的成交明细
// $parms = array('entrustSheetId'=>14023097);
// $res = $bitz->getContractTradeResult($parms);

//获取我的成交历史
// $parms = array('contractId'=>101,'page'=>1,'pageSize'=>50);
// $res = $bitz->getContractMyTrades($parms);

// var_dump($res);die;

File: bitz_java_sdk/BitzApi.java

import java.io.IOException;
import java.util.*;

public class BitzApi {

    private String apiKey;
    private String secretKey;
    private String tradePwd;
    private String WEB_BASE; //


    public BitzApi(String api_server, String api_key, String secret_key) {
        this.WEB_BASE = api_server;
        this.apiKey = api_key;
        this.secretKey = secret_key;
        this.tradePwd = "";
    }


    /**
     * 获取合约账户信息
     *
     * @return String
     */
    public String getAccountInfo() throws IOException {

        HashMap<String, String> params = new HashMap<String, String>();
        return this.sign_api("/Contract/getContractAccountInfo", params);
    }


    /**
     * 下单(做多/做空)
     *
     * @param contract_id 合约ID
     * @param price       价格
     * @param amount      合约张数
     * @param leverage    杠杆倍数x
     * @param direction   委托方向
     * @param type        委托类型
     * @param is_cross    是否全仓
     */

    public String getPlaceOrder(String contract_id, String price, String amount, String leverage, String direction, String type, String is_cross) throws IOException {

        HashMap<String, String> params = new HashMap<String, String>();
        params.put("contractId", contract_id);
        params.put("amount", amount);
        params.put("leverage", leverage);
        params.put("direction", direction);
        params.put("type", type);
        params.put("isCross", is_cross);
        if (!"market".equals(type)) {
            params.put("price", price);
        }
        return this.sign_api("/Contract/addContractTrade", params);
    }


    /**
     * 取消委托
     * order_id  委托单id
     */

    public String getCancelOrder(String order_id) throws IOException {

        HashMap<String, String> params = new HashMap<String, String>();
        params.put("entrustSheetId", order_id);
        return this.sign_api("/Contract/cancelContractTrade", params);

    }


    /**
     * 获取我的活动委托
     * contract_id  合约ID
     */

    public String getActiveOrders(String contract_id) throws IOException {

        HashMap<String, String> params = new HashMap<String, String>();
        params.put("contractId", contract_id);
        return this.sign_api("/Contract/getContractOrder", params);

    }


    /**
     * 获取当前未平仓位
     * contract_id  合约ID
     */

    public String getMyActivePositions(String contract_id) throws IOException {

        HashMap<String, String> params = new HashMap<String, String>();
        params.put("contractId", contract_id);
        return this.sign_api("/Contract/getContractActivePositions", params);
    }


    /**
     * 获取某个合约交易的市场深度
     *
     * @param contract_id 合约ID
     * @return String
     */

    public String getMxDataOrderBook(String contract_id) throws IOException {

        Map<String, String> params = new HashMap<String, String>();
        params.put("contractId", contract_id);
        return this.market_api("/Market/getContractOrderBook", params);
    }


    /**
     * 获取某个合约交易最新行情
     *
     * @param contract_id (合约ID)
     * @return String
     */

    public String getMxDataTickers(String contract_id) throws IOException {

        Map<String, String> params = new HashMap<String, String>();
        params.put("contractId", contract_id);
        return this.market_api("/Market/getContractTickers", params);
    }


    /**
     * 获取某个合约交易历史
     *
     * @param contract_id 合约ID
     * @param page_size
     * @return
     * @throws IOException
     */
    public String getMxDataTrades(String contract_id, String page_size) throws IOException {

        Map<String, String> params = new HashMap<String, String>();
        params.put("contractId", contract_id);
        params.put("page_size", page_size);
        return this.market_api("/Market/getContractTradesHistory", params);

    }

    /**
     * 获取某个合约交易k线
     *
     * @param contract_id 合约ID
     * @param type        K线类型(5m, 15m, 30m, 1h, 4h, 8h, 1d)
     * @param amount      获取数据的数量
     * @return
     * @throws IOException
     */

    public String getMxDatakData(String contract_id, String type, String amount) throws IOException {

        Map<String, String> params = new HashMap<String, String>();
        params.put("contractId", contract_id);
        params.put("type", type);
        params.put("size", amount);
        return this.market_api("/Market/getContractKline", params);

    }


    /**
     * 获取我的委托历史
     * contract_id  合约ID
     */

    public String getMyOrders(String contract_id) throws IOException {

        HashMap<String, String> params = new HashMap<String, String>();
        params.put("contractId", contract_id);
        return this.sign_api("/Contract/getContractMyHistoryTrade", params);

    }


    /**
     * 获取我的已平仓位
     * contract_id  合约ID
     */

    public String getMyPositions(String contract_id, String page, String pageSize) throws IOException {

        HashMap<String, String> params = new HashMap<String, String>();
        params.put("contractId", contract_id);
        params.put("page", page);
        params.put("pageSize", pageSize);
        return this.sign_api("/Contract/getContractMyPositions", params);

    }


    /**
     * 获取某个委托的成交明细
     * order_id  委托单ID
     */

    public String getOrderMatchResults(String order_id) throws IOException {

        HashMap<String, String> params = new HashMap<String, String>();
        params.put("entrustSheetId", order_id);
        return this.sign_api("/Contract/getContractTradeResult", params);

    }


    /**
     * 获取合约交易市场列表
     *
     * @param contract_id (合约id)  必传 false
     */
    public String getMxDataAontracts(String contract_id) throws IOException {

        Map<String, String> params = new HashMap<String, String>();
        params.put("contractId", contract_id);
        return this.market_api("/Market/getContractCoin", params);
    }


    /**
     * 获取合约交易的成交历史
     * contract_id  合约ID
     */

    public String getMyTrades(String contract_id) throws IOException {

        Map<String, String> params = new HashMap<String, String>();
        params.put("contractId", contract_id);
        return this.market_api("/Market/getContractTradesHistory", params);
    }


    /**
     * @param path
     * @param params
     * @return
     */
    private String market_api(String path, Map<String, String> params) {
        String result = "";
        String strParams = SignUtil.createLinkString(params);
        try {
            result = HttpUtilManager.getInstance().requestHttpGet(WEB_BASE, path, strParams);
        } catch (Exception e) {
            result = "";
        }
        return result;
    }

    private String sign_api(String path, Map<String, String> params) {
        String result = "";

        long timeStamp = System.currentTimeMillis();
        timeStamp = timeStamp / 1000;
        params.put("apiKey", this.apiKey);
        params.put("timeStamp", String.valueOf(timeStamp));
        params.put("nonce", String.valueOf(timeStamp).substring(0, 6));
        String sign = SignUtil.buildSign(params, this.secretKey);
        params.put("sign", sign);
        //
        try {
            result = HttpUtilManager.getInstance().requestHttpPost(WEB_BASE, path, params);
        } catch (Exception e) {

        }
        return result;
    }


}
